### PR TITLE
fix 'embedding' frontend function for torch and tf

### PR DESCRIPTION
**Pull request by Sam-Armstrong**
The current 'embedding' torch frontend function works by looping through every single value in the indices tensor and embedding them. This is a problem because we need the batch size to be variable in order for the transpiled graph to run on an input of any given batch size. This problem was identified by a discrepancy in the output values of ImageGPT when it had been transpiled from torch to jax.

I've fixed this by calling into the newly fixed ivy.embedding function (https://github.com/unifyai/ivy/pull/17154). The torch frontend also has the ability to perform its own re-norm depending on the values of max_norm and norm_type. Also fixed the embedding frontend tests for both torch and tensorflow.

Depends on https://github.com/unifyai/ivy/pull/17154 being merged.

### CODE REVIEW

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_nn.py [hunk 1]
```diff
@@ -1340,7 +1340,7 @@ def test_tensorflow_softmax(
 @handle_frontend_test(
     fn_tree="tensorflow.nn.embedding_lookup",
     dtypes_indices_weights=helpers.embedding_helper(),
-    max_norm=st.floats(min_value=0, max_value=5, exclude_min=True),
+    max_norm=st.floats(min_value=0.1, max_value=5, exclude_min=True),
```
Comment (Sam-Armstrong) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_nn.py:1343]: Values smaller than 0.1 still work fine, but there is slight imprecision when working with very small values which can cause the test to fail.

**Review by AnnaTz — APPROVED** [review]
I don't think there was an `ivy.embedding` when this function was added. This implementation looks much better! 
Feel free to merge if the tests are passing 🙂

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Pull request merged**